src/aria2.cpp

```diff
@@ -195,6 +197,14 @@ std::string Aria2::tellStatus(const std::string& gid, const std::vector<std::str
   return doRequest(methodCall);
 }
 
+std::string Aria2::getNewRpcSecret()
+{
+  unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
+  std::string letters("0123456789abcdefghijklmnopqrstuvqxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+  std::shuffle(letters.begin(), letters.end(), std::default_random_engine(seed));
+  return letters.substr(0, 9);
```

Comment:
This is not much better than simply converting `seed` to a string. The output is fully determined by the `seed` value (unless the underlying random engine also adds some host-dependent salt).

I think that returning the stringified `seed` value provides an adequate level of security. The extra code only creates a false impression of higher security.

Reply:
@veloman-yunkan  I don't understand your comment. `seed` is the clock, so it is not random and can not be used as random value. The shuffled string is not directly depending on `seed` but on the pseudo random generator of `std::default_random_engine` (itself seeded by `seed`). So, "yes" this is pseudo-random, but this is the usual way of doing to get a random value.

Reply:
@kelson42 As long as you know the value of `seed` you can find out the result of `std::shuffle()`, can't you?

Reply:
@veloman-yunkan Now I understand your comment. To answer your question, you might know or not. I don't know the details of implementation of the pseudo-random generator. But I still prefer this way as just putting the unix date (even if I understand your point). Maybe you have an idea to easily improve the quality of the seed?

Reply:
> But I still prefer this way as just putting the unix date

That's my concern. The result of the `shuffle()` is no better than the timestamp used as a seed value. If you are uncomfortable with the unix date returned directly you can reverse or square it (but that is intended only to illustrate my point).

> Maybe you have an idea to easily improve the quality of the seed?

I don't believe that this is the case where we should go into such lengths. Security measures should be proportional to the risks. I think that we have a lot of similar vulnerabilities in other places.

Reply:
BTW, why not taking advantage of the [`gen_uuid()`](https://github.com/kiwix/libkiwix/blob/b70c92cadefdb10295ad8bb34c992407519d193a/src/tools/otherTools.cpp#L368) function from `src/tools/otherTools.cpp`?

Reply:
Does `gen_uuid()` not suffer from the same problem of creating a false impression?
If we can get the random string, we can also get its uuid?

Reply:
If called with an empty string, `gen_uuid()` generates a random UUID based on the wall time and process CPU time values. Even though it is not an ideal source of random strings, the advantage of reusing it is that we don't create a new way of producing random strings. If we ever decide that our random string generation must be improved we will have to do it in a single place. 

Note that you may need to drop the dashes from the result (if they are not allowed in the token strings accepted by aria2).